json-format.md

```diff
@@ -129,34 +129,82 @@ OPTIONAL not omitted attributes MAY be represeted as a `null` JSON value.
 
 ### 3.1. Handling of "data"
 
+The JSON representation of the event "data" payload is determined by the runtime
+type of the `data` content and the value of the [`datacontenttype`
+attribute][datacontenttype].
+
+#### 3.1.1. Payload Serialization
+
 Before taking action, a JSON serializer MUST first determine the runtime data
 type of the `data` content.
 
 If the implementation determines that the type of data is `Binary`, the value
 MUST be represented as a [JSON string][json-string] expression containing the
 [Base64][base64] encoded binary value, and use the member name `data_base64` to
-store it inside the JSON object.
-
-For any other type, the implementation MUST translate the data value into a
-[JSON value][json-value], and use the member name `data` to store it inside the
-JSON object.
-
-Out of this follows that use of the `data` and `data_base64` members is mutually
-exclusive in a JSON serialized CloudEvent.
+store it inside the JSON representation. The `datacontenttype` SHOULD reflect
+the format of the original binary data.
+
+If the type of data is NOT `Binary`, the implementation MUST next determine
+whether the value of the `datacontenttype` attribute is _JSON-oriented_. A
+JSON-oriented content type is one whose subtype is `json` or has a `+json`
+format extension—that is, if it has the form `*/json` or `*/*+json`.
+
+If the `datacontenttype` is JSON-oriented, a JSON serializer MUST translate the
+data value to a [JSON value][json-value], and use the member name `data` to
+store it inside the JSON representation. Note: this translation SHOULD NOT
```

Comment:
The issue here is with terminology. We seem to agree that we need to translate (conceptually) object of class `Player` to object of class `JSONObject` (or whatever the appropriate class might be), but _not_ to object of class `String` whose value is the JSON representation. I imagine the latter, not the former, to be what is meant by "serialization", because in my mind, "serialization" results in a string (or at least a byte sequence). But I could be wrong, and at the very least it does seem imprecise. I'd open to suggestions.

Reply:
How about something like:

> Data values which already represent JSON values MUST be serialized directly as those values, rather than being serialized to a JSON document stored in a string.

Still not great, but I think it's a little less worrying.